src/main/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -466,6 +466,32 @@ void enableFileSource(final SamReader reader, final boolean enabled) {
             iterator.setFileSource(enabled ? reader : null);
     }
 
+    /**
+     * Prepare to iterate through SAMRecords that match the given intervals.
```

Comment:
I think we should slightly narrow the description of what this method does. Since there is no validation that the provided chunks actually match (fully span) the territory covered by the intervals, it possible that the iterator doesn't return the full set of records matching the query. The records are constrained first by the chunks, then further constrained by the intervals.

Reply:
Good point. I've revised the wording to say "Prepare to iterate through SAMRecords that match the intersection of the given intervals and chunk boundaries."